static/app/views/insights/mobile/screens/components/screensOverview.tsx

```diff
@@ -53,164 +53,168 @@ function useMetrics(
+  } = secondaryResult;
 
   useEffect(() => {
     if (primaryData) {
       const screens: string[] = [];
-      primaryData?.data.forEach(row => {
+      primaryData?.forEach(row => {
         if (row.transaction) {
           screens.push(String(row.transaction));
         }
       });
-      setVisibleScreens(screens);
+      visibleScreensRef.current = screens;
+      setHasVisibleScreens(screens.length > 0);
     }
   }, [primaryData]);
 
-  const tableSearchFilters = new MutableSearch(['transaction.op:ui.load']);
-  const derivedQuery = getTransactionSearchQuery(location, primaryEventView.query);
-
-  const combinedData = useMemo((): TableData | undefined => {
-    if (defined(primaryData) && defined(secondaryData)) {
-      const meta: MetaType = {};
-      meta.units = {
-        ...secondaryData.meta?.units,
-        ...primaryData.meta?.units,
-      };
-      meta.fields = {
-        ...secondaryData.meta?.fields,
-        ...primaryData.meta?.fields,
-      };
-
-      const data = primaryData.data.map(row => {
-        const matchingRow = secondaryData.data.find(
-          metricRow => metricRow.transaction === row.transaction
-        );
-        if (matchingRow) {
-          return {
-            ...matchingRow,
-            ...row,
-          };
-        }
-        return row;
-      });
-
-      return {
-        data,
-        meta,
-      };
-    }
-
-    return primaryData;
-  }, [primaryData, secondaryData]);
+  const derivedQuery = getTransactionSearchQuery(location, primaryQuery);
+
+  const combinedData = useMemo(() => {
+    const meta: MetaType = {};
+    meta.units = {
+      ...secondaryMeta?.units,
+      ...primaryMeta?.units,
+    };
+    meta.fields = {
+      ...secondaryMeta?.fields,
+      ...primaryMeta?.fields,
+    };
+
+    const data = primaryData.map(row => {
+      const matchingRow = secondaryData.find(
+        metricRow => metricRow.transaction === row.transaction
+      );
+      if (matchingRow) {
+        return {
+          ...matchingRow,
+          ...row,
+        };
+      }
+      return row;
+    });
+
+    return {
+      data,
+      meta,
+    };
+  }, [primaryData, secondaryData, primaryMeta, secondaryMeta]);
 
   const loading = primaryLoading || (hasVisibleScreens && secondaryLoading);
   return (
     <Container>
       <SearchBar
-        eventView={primaryEventView}
         onSearch={search => {
-          router.push({
-            pathname: router.location.pathname,
+          navigate({
+            pathname: location.pathname,
             query: {
               ...location.query,
               cursor: undefined,
               query: String(search).trim() || undefined,
             },
           });
         }}
-        organization={organization}
-        query={getFreeTextFromQuery(derivedQuery)!}
+        query={getFreeTextFromQuery(derivedQuery)}
         placeholder={t('Search for Screen')}
-        additionalConditions={tableSearchFilters}
```

Comment:
I noticed we swapped this to the generic search bar component. Is there a reason we wanted to get away from the performance search bar? The performance search bar would show results that match as you typed and this means the user doesn't get the autocomplete if they're looking for a transaction that isn't on the main page

Reply:
Good question, the perf search bar doesn't work on eap, we have a eap version available that I can add here shortly.